### PR TITLE
Rename Relational Scan to Cross-partition Scan

### PR DESCRIPTION
**Pull request by brfrn169**
## Description

This PR renames Relational Scan to Cross-partition Scan. 

## Related issues and/or PRs

- https://github.com/scalar-labs/scalardb/pull/889
- https://github.com/scalar-labs/scalardb/pull/900
- https://github.com/scalar-labs/scalardb/pull/908
- https://github.com/scalar-labs/scalardb/pull/913
- https://github.com/scalar-labs/scalardb/pull/925

## Changes made

- Renamed `Relational Scan` to `Cross-partition Scan`

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

### CODE REVIEW

File: core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitCrossPartitionScanIntegrationTestWithJdbcDatabase.java [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package com.scalar.db.storage.jdbc;
 
-import com.scalar.db.transaction.consensuscommit.ConsensusCommitRelationalScanIntegrationTestBase;
+import com.scalar.db.transaction.consensuscommit.ConsensusCommitCrossPartitionScanIntegrationTestBase;
```
Comment (brfrn169) [inline core/src/integration-test/java/com/scalar/db/storage/jdbc/ConsensusCommitCrossPartitionScanIntegrationTestWithJdbcDatabase.java:3]: Renamed the integration tests related to cross-partition scan.

File: core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public class SimpleSelectQuery implements SelectQuery {
   private final boolean isRangeQuery;
   private final Optional<String> indexedColumn;
   private final boolean isConditionalQuery;
-  private final boolean isRelationalQuery;
+  private final boolean isCrossPartitionQuery;
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java:40]: (CC: @jnmt) Renamed this flag as well.

File: core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java [hunk 3]
```diff
@@ -106,7 +106,7 @@ private String conditionSqlString() {
     return String.join(" AND ", conditions);
   }
 
-  private String relationalConditionSqlString() {
+  private String crossPartitionConditionSqlString() {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java:109]: Ditto. Renamed this method.

File: core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java [hunk 4]
```diff
@@ -200,7 +200,7 @@ private String orderBySqlString() {
             .collect(Collectors.joining(","));
   }
 
-  private String relationalOrderBySqlString() {
+  private String crossPartitionOrderBySqlString() {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java:203]: Ditto.

File: core/src/test/java/com/scalar/db/storage/jdbc/JdbcServiceTest.java [hunk 5]
```diff
@@ -160,7 +160,8 @@ public void whenGetScannerExecuted_withScanAll_shouldCallQueryBuilder() throws E
   }
 
   @Test
-  public void whenGetScannerExecuted_withRelationalScan_shouldCallQueryBuilder() throws Exception {
+  public void whenGetScannerExecuted_withCrossPartitionScan_shouldCallQueryBuilder()
```
Comment (brfrn169) [inline core/src/test/java/com/scalar/db/storage/jdbc/JdbcServiceTest.java:163]: Renamed some unit test methods.

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Review by jnmt — APPROVED** [review]
LGTM! Thank you for completing a laborious task.

**Pull request merged**